### DISCUSSION TITLE
Difficulty Accessing InvenTree on VPS and Configuring Domain

### DISCUSSION THREAD

**Discussion by guilhermedecastrogt** [inventree/InvenTree#6770] [Q&A]
Hello everyone,

I hope this message finds you well. I'm encountering some issues while trying to access InvenTree on my VPS (Virtual Private Server) and configuring it to work properly. I have another application running on port 8000, and by default, InvenTree also wants to use port 8000. As a result, I'm facing conflicts in accessing both applications simultaneously.

Could anyone guide me through the process of changing the port for InvenTree? I've looked through the documentation but couldn't find a straightforward solution. Any help or pointers in the right direction would be greatly appreciated.

Additionally, I'm interested in setting up a domain for the IP address of my VPS. Could someone provide instructions or resources on how to achieve this? I'm relatively new to managing VPS configurations, so any beginner-friendly explanations would be fantastic.

Thank you all for your time and assistance.

**Comment by SchrodingersGat** [comment]
@guilhermedecastrogt are you running InvenTree under docker? Or bare metal?

**Reply by guilhermedecastrogt** [reply]
Hey @SchrodingersGat,

Thanks for your prompt response. Yes, I'm running InvenTree under Docker. Do you have any insights on how I can address the port conflict issue within the Docker environment? Any guidance would be much appreciated.

Additionally, if you have any tips on configuring a domain for my VPS IP address while running services within Docker containers, I'd be grateful for that as well.

Looking forward to your suggestions!

**Reply by guilhermedecastrogt** [reply]
Regarding the domain setup, I noticed in the documentation that at the end, it suggests accessing InvenTree via http://inventree.localhost/. However, I'm not quite sure how to replace this with the IP address of my VPS.

Could you provide some guidance on how I can configure InvenTree to be accessible via the IP address of my VPS instead of "inventree.localhost"?

**Reply by SchrodingersGat** [reply]
As per the [docker setup guide](https://docs.inventree.org/en/latest/start/docker_install) you will need to adjust the value of `INVENTREE_SITE_URL` from *http://inventree.localhost* to the host address of the server, e.g. 

- *https://192.168.120.10*
- *https://inventree.my-host.com*

**Reply by guilhermedecastrogt** [reply]
But does it only have to change in Docker Composer or does it have to be changed somewhere else? ex cadyfile

Will need an SSL certificate or can it be http://?

**Reply by SchrodingersGat** [reply]
The caddyfile uses the host name that you specify in the `.env` file (don't edit the docker-compose or caddyfile unless you have good reason do, just edit `.env`).

> Will need an SSL certificate or can it be http://?

That really depends on your application. From the perspective of the Caddyfile you can use either. If you use https then the SSL certificate should be added automatically via letsencrypt

**Reply by guilhermedecastrogt** [reply]
But, what is default port to access the inventree if I just put the vps ip?

**Reply by SchrodingersGat** [reply]
You should not have to specify the port for the internal address - it listens on 80 for HTTP or 443 for HTTPS
